Sources/Filters/General/ContourTriangulator/helper.js

```diff
@@ -0,0 +1,2289 @@
+import macro from 'vtk.js/Sources/macros';
+import vtkPoints from 'vtk.js/Sources/Common/Core/Points';
+import * as vtkMath from 'vtk.js/Sources/Common/Core/Math';
+import vtkLine from 'vtk.js/Sources/Common/DataModel/Line';
+import vtkPolygon from 'vtk.js/Sources/Common/DataModel/Polygon';
+import vtkIncrementalOctreePointLocator from 'vtk.js/Sources/Common/DataModel/IncrementalOctreePointLocator';
+import vtkBoundingBox from 'vtk.js/Sources/Common/DataModel/BoundingBox';
+import { CCS_POLYGON_TOLERANCE } from './Constants';
+
+const { vtkErrorMacro } = macro;
+
+/**
+ * Reverse the elements between the indices first and last of the given array arr.
+ *
+ * @param {any[]} arr
+ * @param {Number} first
+ * @param {Number} last
+ */
+export function reverseElements(arr, first, last) {
+  const mid = first + Math.floor((last - first) / 2);
+  for (let i = first; i <= mid; ++i) {
+    [arr[i], arr[last - (i - first)]] = [arr[last - (i - first)], arr[i]];
+  }
+}
+
+// TODO: Document functions
+
+// ---------------------------------------------------
+/**
+ * Compute the normal of a polygon.
+ *
+ * @param {Vector3[]} poly
+ * @param {vtkPoints} points
+ * @param {Vector3} normal
+ * @returns {Number}
+ */
+export function vtkCCSPolygonNormal(poly, points, normal) {
+  const n = poly.length;
+  const nn = [0.0, 0.0, 0.0];
+  const p0 = [];
+  let p1 = [];
+  let p2 = [];
+
+  points.getPoint(poly[0], p0);
+  points.getPoint(poly[1], p1);
+
+  const v1 = [];
+  const v2 = [];
+  for (let j = 2; j < n; j++) {
+    points.getPoint(poly[j], p2);
+    vtkMath.subtract(p2, p1, v2);
+    vtkMath.subtract(p0, p1, v2);
+
+    nn[0] += v1[1] * v2[2] - v1[2] * v2[1];
+    nn[1] += v1[2] * v2[0] - v1[0] * v2[2];
+    nn[2] += v1[0] * v2[1] - v1[1] * v2[0];
+
+    [p1, p2] = [p2, p1];
+  }
+
+  const norm2 = vtkMath.dot(nn, nn);
+  if (norm2 > 0) {
+    const norm = Math.sqrt(norm2);
+    normal[0] = nn[0] / norm;
+    normal[1] = nn[1] / norm;
+    normal[2] = nn[2] / norm;
+  }
```

Comment:
use `vtkMath.normalize()`
it will return `norm` instead of `norm2` but that shouldn't be a pb.